connector_node/risingwave-source-test/src/test/java/com/risingwave/connector/PostgresSourceTest.java

```diff
@@ -0,0 +1,186 @@
+package com.risingwave.connector;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+import com.risingwave.proto.ConnectorServiceProto;
+import io.grpc.Grpc;
+import io.grpc.InsecureChannelCredentials;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import org.junit.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.utility.MountableFile;
+
+public class PostgresSourceTest {
+    private static final Logger logger =
+            LoggerFactory.getLogger(PostgresSourceTest.class.getName());
+
+    private static final PostgreSQLContainer<?> pg =
+            new PostgreSQLContainer<>("postgres:12.3-alpine")
+                    .withDatabaseName("test")
+                    .withUsername("postgres")
+                    .withCommand("postgres -c wal_level=logical -c max_wal_senders=10");
+
+    public static Server connectorServer =
+            ServerBuilder.forPort(ConnectorService.DEFAULT_PORT)
+                    .addService(new ConnectorServiceImpl())
+                    .build();
+
+    public static SourceTestClient testClient =
+            new SourceTestClient(
+                    Grpc.newChannelBuilder(
+                                    "localhost:" + ConnectorService.DEFAULT_PORT,
+                                    InsecureChannelCredentials.create())
+                            .build());
+
+    @BeforeClass
+    public static void init() {
+        // generate orders.tbl test data
+        SourceTestClient.genOrdersTable(10000);
```

Comment:
Would you generate two copy of test data for Mysql and PG?

Reply:
yes it seems so...maybe I can first check the line number before deciding whether to write data or not :)